keras/layers/core.py

```diff
@@ -899,3 +903,182 @@ def get_config(self):
                   "input_dim": self.input_dim}
         base_config = super(MaxoutDense, self).get_config()
         return dict(list(base_config.items()) + list(config.items()))
+
+
+class Lambda(Layer):
+
+    """Lambda layer for evaluating arbitrary function
+
+    Input shape
+    -----------
+    output_shape of previous layer
+
+    Output shape
+    ------------
+    Specified by output_shape argument
+
+    Arguments
+    ---------
+    function - The function to be evaluated. Takes one argument : output of previous layer
+    output_shape - Expected output shape from function. Could be a tuple or a function of the shape of the input
+    """
+
+    def __init__(self, function, output_shape=None, ndim=2):
+        super(Lambda, self).__init__()
+        self.input = ndim_tensor(ndim)
+        py3 = sys.version_info[0] == 3
+        if py3:
+            self.function = marshal.dumps(function.__code__)
+        else:
+            self.function = marshal.dumps(function.func_code)
+        if output_shape is None:
+            self._output_shape = None
+        elif type(output_shape) in {tuple, list} :
+            self._output_shape = tuple(output_shape)
+        else:
+            if py3:
+                self._output_shape = marshal.dumps(output_shape.__code__)
+            else:
+                self._output_shape = marshal.dumps(output_shape.func_code)
+
+    @property
+    def output_shape(self):
+        if self._ouput_shape is None:
+            return self.input_shape
+        elif type(self._output_shape) == tuple:
+            return self._output_shape
```

Comment:
For the `output_shape` argument to be consistent with the `input_shape` constructor argument, it should not include the first dimension (the samples dimension). As such, the return value here would be `(self.input_shape[0],) + self._output_shape`.

This behavior should also be documented.